zkabacus-crypto/src/revlock.rs

```diff
@@ -87,42 +80,100 @@ pub struct RevocationLockCommitment(pub(crate) Commitment<G1Projective>);
 
 /// Commitment randomness corresponding to a [`RevocationLockCommitment`].
 #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
-#[allow(missing_copy_implementations)]
 pub struct RevocationLockBlindingFactor(pub(crate) BlindingFactor);
 
-impl RevocationSecret {
-    /// Create a new, random revocation secret.
+/// An error in producing a revocation pair.
+#[derive(Debug, Copy, Clone, Error)]
+pub enum Error {
+    /// The provided revocation secret does not produce a valid revocation lock.
+    #[error("The revocation secret does not produce a valid revocation lock")]
+    InvalidSecret(),
+
+    /// The provided revocation secret does not produce the input revocation lock.
+    #[error("The revocation secret does not produce the provided revocation lock")]
+    MismatchedPair(),
```

Comment:
OK... after attempting to implement this, this looks like it would require increasing the visibility of the `UncheckedRevocationSecret`struct. After discussion with @marsella, returning the failing revocation secret here seems to have limited value...

Reply:
That makes sense. Thanks for looking into it. In that case, please remove the `()` for these dataless enum variants.